r/vignettes/developers/install_details.Rmd

```diff
@@ -0,0 +1,90 @@
+# How the R package is installed
+
+In order for the `arrow` R package to work, it needs the Arrow C++ library, 
+also known as libarrow.  There are a number of scripts that are triggered 
+when `R CMD INSTALL .` is run and for Arrow users, these should all just work 
+without configuration and pull in the most complete pieces (e.g. official 
+binaries that we host).
+
+An overview of these scripts is shown below:
+
+* `configure` and `configure.win` - these scripts are triggered during
+`R CMD INSTALL .` on non-Windows and Windows platforms, respectively. They
+handle finding the libarrow, setting up the build variables necessary, and
+writing the package Makevars file that is used to compile the C++ code in the R
+package.
+
+* `tools/nixlibs.R` - this script is sometimes called by `configure` on Linux
+(or on any non-windows OS with the environment variable
+`FORCE_BUNDLED_BUILD=true`). This sets up the build process for our bundled
+builds (which is the default on linux) and checks for binaries or downloads 
+libarrow from source depending on dependency availability and build configuration.
+
+* `inst/build_arrow_static.sh` - called by `tools/nixlibs.R` when libarrow is 
+being built.  It builds libarrow for a bundled, static build, and
+mirrors the steps described in the ["Arrow R Developer Guide" vignette](./setup.html)
+This build script is also what is used to generate the prebuilt binaries.
```

Comment:
```suggestion
This build script is also what is used to generate our prebuilt binaries.
```

Could/should we link to somewhere in our docs where we show how to install our binaries?